src/System.Management.Automation/engine/runtime/Operations/MiscOps.cs

```diff
@@ -1595,25 +1604,40 @@ internal static void CheckActionPreference(FunctionContext funcContext, Exceptio
             // set $? to false indicating an error
             context.QuestionMarkVariableValue = false;
 
-            bool anyTrapHandlers = funcContext._traps.Any() && funcContext._traps.Last().Item2 != null;
+            ActionPreference preference = GetErrorActionPreference(context);
 
-            if (!anyTrapHandlers && !ExceptionHandlingOps.NeedToQueryForActionPreference(rte, context))
+            if (IsExceptionNew(rte, context) &&
```

Comment:
`IsExceptionNew` helper method is not needed since it is a simple comparison and only used once.

Reply:
Funny, I added IsExceptionNew because @iSazonov felt the comment I had was unnecessary in a previous code review, and I wanted other devs reading the code to understand that the comparison was used to check if the exception was new. I prefer the inline comparison with the comment though, so I'm putting that back (although I've worded the comment differently).